wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsFileIOClient.java

```diff
@@ -196,58 +194,55 @@ public boolean writeHPCCFileData(byte []  data, String fileName, String targetLa
         boolean success = true;
         log.debug("Attempting to write data to HPCC File: " + fileName);
 
-        if (wsFileIOServiceSoapProxy == null)
-            throw new Exception ("wsFileIOServiceSoapProxy not available!");
-        else
+        getSoapProxy();
+
+        WriteFileDataRequest writefileparams = new WriteFileDataRequest();
+        writefileparams.setAppend(append);
+        writefileparams.setDestDropZone(targetLandingZone);
+        writefileparams.setDestRelativePath(fileName);
+        writefileparams.setOffset(offset);
+
+        int dataindex = 0;
+        int limit = uploadchunksize <= 0 ? defaultUploadChunkSize : uploadchunksize;
+        long payloadsize = 0;
+        long bytesleft = data.length;
+        byte [] subdata = null;
+
+        while (bytesleft > 0)
         {
-            WriteFileDataRequest writefileparams = new WriteFileDataRequest();
-            writefileparams.setAppend(append);
-            writefileparams.setDestDropZone(targetLandingZone);
-            writefileparams.setDestRelativePath(fileName);
-            writefileparams.setOffset(offset);
-
-            int dataindex = 0;
-            int limit = uploadchunksize <= 0 ? defaultUploadChunkSize : uploadchunksize;
-            long payloadsize = 0;
-            long bytesleft = data.length;
-            byte [] subdata = null;
-
-            while (bytesleft > 0)
-            {
-                payloadsize = bytesleft >= limit ? limit : bytesleft;
-                log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
+            payloadsize = bytesleft >= limit ? limit : bytesleft;
+            log.trace("Writing offset: "+dataindex+"\t size: "+((int)dataindex + (int)payloadsize));
 
-                subdata = new byte [(int)payloadsize];
-                for (int i = 0; i < payloadsize ; i++, dataindex++)
-                    subdata[i] = data[dataindex];
+            subdata = new byte [(int)payloadsize];
+            for (int i = 0; i < payloadsize ; i++, dataindex++)
+                subdata[i] = data[dataindex];
 
-                writefileparams.setData(subdata);
-                writefileparams.setAppend(dataindex > 0);
-                writefileparams.setOffset((long)dataindex);
+            writefileparams.setData(subdata);
+            writefileparams.setAppend(dataindex > 0);
+            writefileparams.setOffset((long)dataindex);
 
-                try
+            try
+            {
+                WriteFileDataResponse writeFileDataResponse = wsFileIOServiceSoapProxy.writeFileData(writefileparams);
+                String result = writeFileDataResponse.getResult();
+
+                log.debug(result);
+                //Wish there was a better way
+                if (!result.startsWith("Failed"))
```

Comment:
That does seem brittle. Do we have a ticket to create a better status code?

Reply:
agreed, requested change in the ESP side: https://track.hpccsystems.com/browse/HPCC-21293
but unlikely to change.